### DISCUSSION TITLE
Guidance Needed for Implementing Weakly Compressible Solver on Adaptive Octree Grids with AMReX

### DISCUSSION THREAD

**Discussion by pa-one55** [AMReX-Codes/amrex#4053] [Q&A]
Hello
I'm a final-year Mechanical Engineering student, and I'm working on a project titled "Development of a Weakly Compressible Solver on Adaptive Octree Grids". This involves developing a CFD solver using adaptive octree grids to simulate incompressible flows.

While I have a good grasp of C++ and mechanical principles, I’m new to CFD and AMReX. I’ve been diving into AMReX for a week, but the documentation has been quite overwhelming.

I’m looking for a clear roadmap to guide me through learning AMReX and successfully implementing my project.
Any help or pointers would be greatly appreciated. Thanks in advance!

Best,
Pawan

**Comment by asalmgren** [comment]
We have compressible and incompressible flow solvers you could look at.
But note that AmReX does level-based not octtree reginement.  Will that
still work for you?

Ann Almgren
Senior Scientist; Dept. Head, Applied Mathematics
Pronouns: she/her/hers

**Comment by pa-one55** [comment]
yeah prof said that works,

**Comment by asalmgren** [comment]
Do you want to build compressible or incompressible solver?

Ann Almgren
Senior Scientist; Dept. Head, Applied Mathematics
Pronouns: she/her/hers

**Reply by pa-one55** [reply]
I'll ask my prof and let you know.

**Reply by pa-one55** [reply]
we want to develop incompressible flow solver on staggered grid.

**Reply by zingale** [reply]
It would be helpful to know what your goal is, since this sounds like it is a thesis project.

Is your project to get the experience of writing an incompressible solver yourself, and then applying it to some test problems?

Or do you want an incompressible solver that you can then apply to some applications?

Both are fine projects, but the approach would differ depending on your goals.  If your goal is the latter, then you may just want to start with an existing AMReX incompressible code.

**Reply by pa-one55** [reply]
Hey
I would like to start with the 2nd one, and no its not a thesis project, its just a project we are supposed to do before completing the bachelors degree .

**Reply by WeiqunZhang** [reply]
You can try either incflo or IAMR.

https://github.com/AMReX-Fluids/incflo

https://github.com/AMReX-Fluids/IAMR

**Comment by pa-one55** [comment]
I am getting this error trying to run HeatEquation_Simple from the guided Tutorials


![Screenshot 2024-07-29 203225](https://github.com/user-attachments/assets/22205f68-7b0c-432c-a5ea-8977f98df02a)

**Comment by asalmgren** [comment]
Did you put the name of the inputs file after the executable when you ran
it?

Ann Almgren
Senior Scientist; Dept. Head, Applied Mathematics
Pronouns: she/her/hers

**Comment by pa-one55** [comment]
i had not.
i did and it worked, thanks
Now i am trying to do the Visualization thing, 
Okay i successfully did the Visualization thing.

Thanks
